packages/runtime/container-runtime/src/blobManager.ts

```diff
@@ -104,7 +105,7 @@ interface PendingBlob {
 
 export interface IPendingBlobs { [id: string]: { blob: string; }; }
 
-export class BlobManager {
+export class BlobManager extends EventEmitter {
```

Comment:
I don't think BlobManager is actually in the container-runtime package's public API. So if we want to expose this to apps we'll need to make some more decisions.

Reply:
I guess we could consume the hasPendingBlobs() method from the container runtime. Not sure about moving the emits also there.

Reply:
Can you please clarify user scenario?
In general, we regret cases like that - places where we exposed "global" observation points about the system. Ideally, each upload is treated separately, and reports its progress to consumer. There will be exceptions, of course, but it would be great to clarify the need.

I have a feeling that it's similar how we expose "readonly" state of container. We started as a boolean and switched later to be a structure, exposing details (is it readonly because user has no write permissions? Or is this preservation file? Or host switched container to read-only?).
I feel you will eventually need something like that for Container's dirty flag and start exposing nuances of what "dirty" might mean, i.e. is there local changes in data model that are not reflected in storage (today's definition of "dirty")? Or some REST call in flight (or consensus op in flight) that is not reflected in local data model, but someone wants to know if there are any such things?

Reply:
Not sure I understand why you brought up the dirty flag in this case. After more discussion and your comments in previous PR we decided that for now we could just have an independent signal for pending upload blob scenario given all concerns you mentioned about not being consistent with dirty state due to the async nature of blob upload.
This is an extract of a chat I had with @Abe27342: "that in practice applications generally care if the user does things like try to exit while a blob is pending upload. An app that just calls uploadBlob where it needs it would currently be forced to implement a centralized layer on top of the blob API to track which blobs they've uploaded, which is a lot of hassle without good reason imo."